File: C++/447. Number of Boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<pair<int, int>>& points) {
        int len = points.size();
        unordered_map<int, int> maple;
        int ans = 0;
        for(int i=0;i<len;i++)
        {
            maple.clear();
            for(int j=0;j<len;j++)
            {
                if(i==j) continue;
                int dx = (points[i].first-points[j].first);
                int dy = (points[i].second-points[j].second);
                int tmp = dx*dx+dy*dy;
                maple[tmp]++;
            }
            
            for(auto &m: maple)
            {
                if(m.second >= 2)
                {
                    int tmp = m.second;
                    ans += tmp*(tmp-1);
                }
            }
        }
        return ans;
    }
};

File: Python/447. Number of Boomerangs.py
class Solution:
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        ans = 0
        for point1 in points:
            maple = dict()
            for point2 in points:
                if point1 == point2:
                    continue
                dx = (point1[0]-point2[0])*(point1[0]-point2[0])
                dy = (point1[1]-point2[1])*(point1[1]-point2[1])
                maple[dx+dy] = maple.get(dx+dy, 0) +1
            for k in maple:
                if maple[k] > 1:
                    ans += maple[k]*(maple[k]-1)
        return ans